git-remote-gosh/src/git_helper/push/mod.rs

```diff
@@ -321,141 +323,180 @@ where
 
         let mut push_handlers: JoinSet<anyhow::Result<()>> = JoinSet::new();
 
-        for line in out.lines() {
-            match line.split_ascii_whitespace().next() {
+        let repo = self.local_repository();
+        let id = repo
+            .rev_parse_single(local_ref)
+            .context("Only single revisions are currently supported")?;
+
+        let new_commit_id = id
+            .object()?
+            .peel_to_kind(git_object::Kind::Commit)
+            .context("Need commitish as starting point")?
+            .id
+            .attach(&repo);
+
+        let mut oids = Vec::new();
+        for commit in new_commit_id.ancestors().all()? {
```

Comment:
here's the problem. it's not `rev-list --objects <ref> ^<commit>`

we need to find all objects in the path from the current HEAD till the last pushed commit